src/games/strategy/triplea/ai/proAI/ProNonCombatMoveAI.java

```diff
@@ -148,6 +146,8 @@ public ProNonCombatMoveAI(final ProAI ai) {
           break;
         }
       }
+    } else {
+      moveUnitsToBestTerritories();
```

Comment:
Np. Its a little rough but once you read through the loop its mostly straightforward. Understanding the break condition is the key. The log levels are different as info is only used as high level to follow what the AI is checking and debug is used for info under that. 

I really didn't read through the loop at all. Only thing I needed to understand for the change was the null check and see it did nothing if a capital doesn't exist. Also I plan to rework this loop eventually anyways to expand its scope.